---
title: "SynCom Data Prep"
author: "Oriane Moyne"
date: "10/19/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Samples description

Multi-omics: metagenomics (metaG), metatranscriptomics (metaT), metaRibo-Seq (metaRS) analysis of 4-days (T1) and 7-days (T2) growth SynCom.  

- X1 and X2 samples: Full SynCom (16 strains)  
- X3: SynCom - _Lysobacter_ dropout
- X4: SynCom - _Mucilaginibacter_ dropout
- X5: SynCom - _Burkholderia_ dropout
- X6: SynCom - _Burkholderia_, _Rhizobium_, _Rhodocccus_, _Brevibacillus_ dropout.


## Load libraries

```{r, message = FALSE}
# Packages I commonly use
library(stringr)
library(ggplot2)
library(FactoMineR)
library(factoextra)
library(reshape)
library(reshape2)
library(gplots)
library(edgeR)
```

## Import data

```{r}
# Count tables with taxonomy annotations (Woltka gotu output)
syncom <- read.table("/home/moyne/Moyne_disk/Soil_project/Final_datasets/FINAL_SynCom_modifs_multiomics_count_table_formatted.tsv", 
                    sep="\t", 
                    dec=".", 
                    header = TRUE, 
                    stringsAsFactors = FALSE, 
                    na.strings = "", 
                    check.names = FALSE, 
                    comment.char = "")

# Correct column names for reproducibility
colnames(syncom) <- str_replace(colnames(syncom), "-", "_")
colnames(syncom) <- str_replace(colnames(syncom), "RNA", "metaT")
colnames(syncom) <- str_replace(colnames(syncom), "Riboseq", "metaRS")
colnames(syncom) <- str_replace(colnames(syncom), "_S_", "_T1_")
```

## Taxonomy Assignment

```{r}
genomes_ID <- str_split(syncom$FeatureID, "_", simplify = TRUE)[,1]
table(genomes_ID) # 18 strains

syncom$genus_strain <- NA
syncom[str_detect(syncom$FeatureID, "MDNCIGFF"), ]$genus_strain <- "Arthrobacter_OAP107"
syncom[str_detect(syncom$FeatureID, "MLEKFOEI"), ]$genus_strain <- "Brevibacillus_OAP136"
syncom[str_detect(syncom$FeatureID, "FOCFADBO"), ]$genus_strain <- "Mucilaginibacter_OAE612"
syncom[str_detect(syncom$FeatureID, "IKHJAPOO"), ]$genus_strain <- "Variovorax_OAS795"
syncom[str_detect(syncom$FeatureID, "LLHGGPMF"), ]$genus_strain <- "Bacillus_OAE603"
syncom[str_detect(syncom$FeatureID, "BKKPEFGL"), ]$genus_strain <- "Bosea_OAE506"
syncom[str_detect(syncom$FeatureID, "FKKIHNNJ"), ]$genus_strain <- "Bradyrhizobium_OAE829"
syncom[str_detect(syncom$FeatureID, "JOPPLEBF"), ]$genus_strain <- "Burkholderia_OAS925"
syncom[str_detect(syncom$FeatureID, "ACPHPGFJ"), ]$genus_strain <- "Chitinophaga_OAE865"
syncom[str_detect(syncom$FeatureID, "JENONDHJ"), ]$genus_strain <- "Niastella_OAS944"
syncom[str_detect(syncom$FeatureID, "PEBMFGFJ"), ]$genus_strain <- "Lysobacter_OAE881"
syncom[str_detect(syncom$FeatureID, "HFAHJMJN"), ]$genus_strain <- "Marmoricola_OAE513"
syncom[str_detect(syncom$FeatureID, "CKEEMGOL"), ]$genus_strain <- "Methylobacterium_OAE516"  # OAE5165 ?
syncom[str_detect(syncom$FeatureID, "MOEDGFBE"), ]$genus_strain <- "Mycobacterium_OAE908"
syncom[str_detect(syncom$FeatureID, "GDBBDLJH"), ]$genus_strain <- "Paenibacillus_OAE614"
syncom[str_detect(syncom$FeatureID, "NBPBNKAL"), ]$genus_strain <- "Rhizobium_OAE497"
syncom[str_detect(syncom$FeatureID, "NBJMGFPL"), ]$genus_strain <- "Rhodococcus_OAS809"
syncom[str_detect(syncom$FeatureID, "NCLMAKNM"), ]$genus_strain <- "Terrabacter_OAE601"

syncom <- syncom[, c(1, ncol(syncom), 2:(ncol(syncom)-1))]
```


```{r}
syncom[, colnames(syncom)=="1_2_T2_metaG"] <- 0
```


## RPKM

Import saf file (contains genomes lengths)

```{r}
saf <- read.table("/home/moyne/Moyne_disk/Soil_project/Final_datasets/Final_18_strains_with_strainN.saf", 
                    sep="\t", 
                    dec=".", 
                    header = FALSE, 
                  col.names = c("GeneID",	"Chr", 	"Start",	"End",	"Strand"),
                    stringsAsFactors = FALSE, 
                    na.strings = "", 
                    check.names = FALSE, 
                    comment.char = "")

saf$GeneID <- gsub("_gene", "", saf$GeneID)

saf$gene_len <- saf$End - saf$Start

saf[order(saf$gene_len), ]

summary(saf$gene_len)
```

```{r}
syncom <- merge(saf[, c(1, ncol(saf))], syncom, by.x = "GeneID", by.y = "FeatureID")
```


# Data Cleaning

```{r}
# Information extracted from ffn file
metadata <- read.delim("/home/moyne/Moyne_disk/Soil_project/Final_datasets/18_strains_genenames.txt", 
                    sep="\t", 
                    dec=".", 
                    header = FALSE, 
                    stringsAsFactors = FALSE, 
                    na.strings = "", 
                    check.names = FALSE, 
                    comment.char = "")

metadata$gene <- str_sub(metadata$V1, 2, 15)
metadata$gene_name <- str_sub(metadata$V1, 17)

metadata <- metadata[, c("gene", "gene_name")]
```


```{r}
syncom <- merge(metadata, syncom, by.x = "gene", by.y = "GeneID")
```


# Filtering

Filter out metaRS reads that were not detected in metaT data.  
This step is optional.


```{r, eval = TRUE}
# Create separate matrices for each omic
metaG_mat <- syncom[, str_detect(colnames(syncom), "metaG")]
colnames(metaG_mat) <- str_replace(colnames(metaG_mat), "_metaG", "")
metaT_mat <- syncom[, str_detect(colnames(syncom), "metaT")]
colnames(metaT_mat) <- str_replace(colnames(metaT_mat), "_metaT", "")
metaRS_mat <- syncom[, str_detect(colnames(syncom), "metaRS")]
colnames(metaRS_mat) <- str_replace(colnames(metaRS_mat), "_metaRS", "")

# order them the same
metaT_mat <- metaT_mat[, colnames(metaG_mat)]
metaRS_mat <- metaRS_mat[, colnames(metaG_mat)]

# Rename columns
colnames(metaG_mat) <- paste(colnames(metaG_mat), "metaG", sep = "_")
colnames(metaT_mat) <- paste(colnames(metaT_mat), "metaT", sep = "_")
colnames(metaRS_mat) <- paste(colnames(metaRS_mat), "metaRS", sep = "_")


# metaG averages (remove metaT and metaRS if absent of BOTH metaG replicates)
metaG_av <- data.frame(t(metaG_mat))
metaG_av$Community <- str_split(rownames(metaG_av), "_", simplify = TRUE)[, 1]
metaG_av$Time <- str_split(rownames(metaG_av), "_", simplify = TRUE)[, 3]

# Filtering
table(is.na(metaG_mat))
table(metaG_mat == 0)
table(metaT_mat == 0)
table(metaRS_mat == 0)

#metaT_mat[is.na(metaG_mat)] <- 0
#metaT_mat[metaG_mat == 0] <- 0
#metaRS_mat[metaG_mat == 0] <- 0
metaRS_mat[metaT_mat == 0] <- 0
#metaT_mat[metaRS_mat == 0] <- 0

syncom <- data.frame(cbind(syncom[, 1:4], metaG_mat, metaT_mat, metaRS_mat))
```

# Data filtering

Negative controls:  

_Terrabacter_ and _Bacillus_ are present in the genome database, but absent from the SynCom (negative controls). They should have the lowest read number in the community.  


```{r, eval = TRUE}
syncom_agg <- syncom[, c(4:ncol(syncom))]
syncom_agg <- aggregate(syncom_agg[, 2:ncol(syncom_agg)], by = list(syncom_agg$genus_strain), sum)

# CPM normalization
syncom_agg_cpm <- syncom_agg
syncom_agg_cpm[, 2:ncol(syncom_agg_cpm)] <- apply(syncom_agg[, 2:ncol(syncom_agg)], 2, function(x) x/sum(x) * 1000000)

# average metaG abundance
syncom_agg_cpm$average_metaG <- apply(syncom_agg_cpm[, str_detect(colnames(syncom_agg_cpm), "metaG")], 1, function(x) mean(x, na.rm = TRUE))

syncom_agg_cpm <- syncom_agg_cpm[order(syncom_agg_cpm$average_metaG), ]
syncom_agg_cpm[, c(1, ncol(syncom_agg_cpm))]
```

Below 50 CPM, and below the lowest abundant microbe of the SynCom, confirms negative controls. We can remove these reads.   


```{r}
# Remove Terrabacter and Bacillus reads
syncom[syncom$genus_strain == "Terrabacter_OAE601", ][, 5:ncol(syncom)] <- 0
syncom[syncom$genus_strain == "Bacillus_OAE603", ][, 5:ncol(syncom)] <- 0

syncom <- syncom[syncom$genus_strain != "Terrabacter_OAE601" & syncom$genus_strain != "Bacillus_OAE603", ]

# Remove removed species from dropout experiments
# syncom[syncom$genus_strain == "Terrabacter_OAE601", !str_detect(colnames(syncom), "2_1")|!str_detect(colnames(syncom), "2_2")][, 5:ncol(syncom)] <- 0
syncom[syncom$genus_strain == "Lysobacter_OAE881", str_detect(colnames(syncom), "3_1")|str_detect(colnames(syncom), "3_2")] <- 0
syncom[syncom$genus_strain == "Mucilaginibacter_OAE612", str_detect(colnames(syncom), "4_1")|str_detect(colnames(syncom), "4_2")] <- 0
syncom[syncom$genus_strain == "Burkholderia_OAS925", str_detect(colnames(syncom), "5_1")|str_detect(colnames(syncom), "5_2")] <- 0
syncom[syncom$genus_strain %in% c("Burkholderia_OAS925", "Rhizobium_OAE497", "Rhodococcus_OAS809", "Brevibacillus_OAP136"), str_detect(colnames(syncom), "6_1")|str_detect(colnames(syncom), "6_2")] <- 0
```


## Export non-normalized table

```{r}
syncom_no_norm <- syncom # save non-normalized table
```


```{r, eval = FALSE}
write.table(syncom_no_norm, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/Syncom_formatted_filtered_corrected_no_norm_metaG_filtered.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)
write.table(syncom_no_norm, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/Syncom_formatted_filtered_corrected_no_norm.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)
```

## RPKM normalization

```{r}
rpkm_fun <- function(x) x / ( (syncom$gene_len/1000) * (sum(x)/1000000))
syncom[, 5:ncol(syncom)] <- apply(syncom[, 5:ncol(syncom)], 2, rpkm_fun)    
```

Filter out very low counts (optional)

```{r, eval = FALSE}
syncom[, 5:ncol(syncom)][syncom[, 5:ncol(syncom)] < 1] <- 0 # Replace if < 1 RPKM (optional)
```

## Barplot sequencing depth after normalization

```{r, fig.width = 9, fig.height = 4}
apply(syncom[, 5:ncol(syncom)], 2, sum)
barplot(apply(syncom[, 5:ncol(syncom)][, str_detect(colnames(syncom[, 5:ncol(syncom)]), "metaG")], 2, sum), las = 2)
barplot(apply(syncom[, 5:ncol(syncom)][, str_detect(colnames(syncom[, 5:ncol(syncom)]), "metaT")], 2, sum), las = 2)
barplot(apply(syncom[, 5:ncol(syncom)][, str_detect(colnames(syncom[, 5:ncol(syncom)]), "metaRS")], 2, sum), las = 2)

barplot(apply(syncom[, 5:ncol(syncom)], 2, sum), las = 2)
```

## Export RPKM-normalized table

```{r, eval = FALSE}
write.table(syncom, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/Syncom_formatted_filtered_corrected_RPKM.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)
write.table(syncom, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/Syncom_formatted_filtered_corrected_RPKM_metaG_filtered.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)
```


#--------------------------------

# KEGG Annotation

## Data import (KEGG annotated genomes)

Genomes have been KEGG annotated using BlastKOALA.  

```{r, eval = TRUE}
genes_KEGG1 <- read.table("/home/moyne/Moyne_disk/Soil_project/Final_datasets/18_genomes_KEGG_BlastKOALA.csv", 
                    sep=",", 
                    dec=".", 
                    header = TRUE, 
                    stringsAsFactors = FALSE, 
                    na.strings = "", 
                    check.names = FALSE, 
                    comment.char = "")


dim(genes_KEGG1) # 111949 genes
genes_KEGG1 <- genes_KEGG1[!is.na(genes_KEGG1$Category), ]
nrow(genes_KEGG1) / 111949 * 100 # 23.6% genes annotated

genes_KEGG1[order(genes_KEGG1$gene), ]

genes_KEGG1 <- genes_KEGG1[, c(2:ncol(genes_KEGG1))]

rmarkdown::paged_table(head(genes_KEGG1))

genes_KEGG1[str_detect(genes_KEGG1$pathway, "tryptophan.=>.rebe"), ]
genes_KEGG1[str_detect(genes_KEGG1$kegg, "K00588"), ]

genes_KEGG <- genes_KEGG1
```

## Merge count table with KEGG annotation 

```{r}
syncom_KEGG <- merge(genes_KEGG, syncom, by.x = "gene", by.y = "gene", all.y = TRUE)
```

# Barplot : stay similar. good

```{r, fig.width=15, fig.height=6}
barplot(apply(syncom[,5:ncol(syncom)], 2, sum), las = 2)
barplot(apply(syncom_KEGG[,9:ncol(syncom_KEGG)], 2, sum), las = 2)
```

## Export tables

```{r, eval = FALSE}
write.table(syncom_KEGG, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/Syncom_formatted_filtered_corrected_RPKM_KEGG.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)

syncom_KEGG <- syncom_KEGG[!is.na(syncom_KEGG$kegg), ]

write.table(syncom_KEGG, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/Syncom_formatted_filtered_corrected_RPKM_KEGG_no_NA.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)
```
