---
title: "SynCom Data Prep"
author: "Oriane Moyne"
date: "10/19/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load libraries

```{r, message = FALSE}
# Packages I commonly use
library(stringr)
library(ggplot2)
library(FactoMineR)
library(factoextra)
library(reshape)
library(reshape2)
library(gplots)
#library(edgeR)
```

# Import data

## Data import (Count table)

```{r}
# Count tables with taxonomy annotations (Woltka gotu output)
syncom <- read.table("/home/moyne/Moyne_disk/Soil_project/Final_datasets/PROBIOTOCS_EXPT_3/Dropout_expt2_Syncom_metaG_count_table.tsv", 
                    sep="\t", 
                    dec=".", 
                    header = TRUE, 
                    stringsAsFactors = FALSE, 
                    na.strings = "", 
                    check.names = FALSE, 
                    comment.char = "")
```

## Taxonomy Assignment

```{r}
genomes_ID <- str_split(syncom$FeatureID, "_", simplify = TRUE)[,1]
sort(table(genomes_ID)) # 18 strains

syncom$genus_strain <- NA
syncom[str_detect(syncom$FeatureID, "MDNCIGFF"), ]$genus_strain <- "Arthrobacter_OAP107"
syncom[str_detect(syncom$FeatureID, "MLEKFOEI"), ]$genus_strain <- "Brevibacillus_OAP136"
syncom[str_detect(syncom$FeatureID, "FOCFADBO"), ]$genus_strain <- "Mucilaginibacter_OAE612"
syncom[str_detect(syncom$FeatureID, "IKHJAPOO"), ]$genus_strain <- "Variovorax_OAS795"
syncom[str_detect(syncom$FeatureID, "LLHGGPMF"), ]$genus_strain <- "Bacillus_OAE603"
syncom[str_detect(syncom$FeatureID, "BKKPEFGL"), ]$genus_strain <- "Bosea_OAE506"
syncom[str_detect(syncom$FeatureID, "FKKIHNNJ"), ]$genus_strain <- "Bradyrhizobium_OAE829"
syncom[str_detect(syncom$FeatureID, "JOPPLEBF"), ]$genus_strain <- "Burkholderia_OAS925"
syncom[str_detect(syncom$FeatureID, "ACPHPGFJ"), ]$genus_strain <- "Chitinophaga_OAE865"
syncom[str_detect(syncom$FeatureID, "JENONDHJ"), ]$genus_strain <- "Chitinophagaceae_OAS944"
syncom[str_detect(syncom$FeatureID, "PEBMFGFJ"), ]$genus_strain <- "Lysobacter_OAE881"
syncom[str_detect(syncom$FeatureID, "HFAHJMJN"), ]$genus_strain <- "Marmoricola_OAE513"
syncom[str_detect(syncom$FeatureID, "CKEEMGOL"), ]$genus_strain <- "Methylobacterium_OAE515"  # OAE515 ?
syncom[str_detect(syncom$FeatureID, "MOEDGFBE"), ]$genus_strain <- "Mycobacterium_OAE908"
syncom[str_detect(syncom$FeatureID, "GDBBDLJH"), ]$genus_strain <- "Paenibacillus_OAE614"
syncom[str_detect(syncom$FeatureID, "NBPBNKAL"), ]$genus_strain <- "Rhizobium_OAE497"
syncom[str_detect(syncom$FeatureID, "NBJMGFPL"), ]$genus_strain <- "Rhodococcus_OAS809"
syncom[str_detect(syncom$FeatureID, "NCLMAKNM"), ]$genus_strain <- "Terrabacter_OAE601"

sort(table(syncom$genus_strain)) # 18 strains. Lowest coverages are negative controls Terrabacter and Bacillus (see below)

syncom <- syncom[, c(1, ncol(syncom), 2:(ncol(syncom)-1))]
```

```{r}
# Remove Terrabacter and Bacillus reads (negative controls - were present in the genome index but NOT added to the SynCom)
syncom[syncom$genus_strain == "Terrabacter_OAE601", ][, 4:ncol(syncom)] <- 0
syncom[syncom$genus_strain == "Bacillus_OAE603", ][, 4:ncol(syncom)] <- 0
```

## Barplot sequencing depth

```{r, fig.width = 15, fig.height = 5}
sort(apply(syncom[, 3:ncol(syncom)], 2, sum))
```

## Discard bad sample (did not amplify properly - very low count) 

```{r, eval = TRUE}
syncom <- syncom[, !str_detect(colnames(syncom), "S84")]  # 3_2. We still have 2 replicates remaining
```

## RPKM normalization

Import saf file

```{r}
saf <- read.table("/home/moyne/Moyne_disk/Soil_project/Final_datasets/Final_18_strains_with_strainN.saf", 
                    sep="\t", 
                    dec=".", 
                    header = FALSE, 
                    col.names = c("GeneID",	"Chr", 	"Start",	"End",	"Strand"),
                    stringsAsFactors = FALSE, 
                    na.strings = "", 
                    check.names = FALSE, 
                    comment.char = "")

saf$GeneID <- gsub("_gene", "", saf$GeneID)

saf$gene_len <- saf$End - saf$Start

saf[order(saf$gene_len), ]

summary(saf$gene_len)
```

```{r}
syncom <- merge(saf[, c(1, ncol(saf))], syncom, by.x = "GeneID", by.y = "FeatureID")
```


## RPKM normalization

```{r}
rpkm_fun <- function(x) x / ( (syncom$gene_len/1000) * (sum(x)/1000000))
syncom[, 4:ncol(syncom)] <- apply(syncom[, 4:ncol(syncom)], 2, rpkm_fun)    
```

## Barplot sequencing depth after normalization

```{r, fig.width = 9, fig.height = 4}
sort(apply(syncom[, 4:ncol(syncom)], 2, sum))
barplot(apply(syncom[, 4:ncol(syncom)], 2, sum), las = 2)
```

## Export RPKM-normalized table

```{r, eval = FALSE}
write.table(syncom, "/home/moyne/Moyne_disk/Soil_project/Final_datasets/PROBIOTOCS_EXPT_3/Dropout_expt2_Syncom_formatted_filtered_corrected_RPKM.tsv", sep = "\t", col.names = TRUE, row.names = FALSE)
```



